Sources/KituraContracts/CodableQuery/QueryEncoder.swift

```diff
@@ -21,23 +21,60 @@ extension CharacterSet {
     static let customURLQueryAllowed = CharacterSet(charactersIn: "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~=:&")
 }
 
-/// Query Parameter Encoder
+/**
+ Query Parameter Encoder.
+ 
+ ### Usage Example: ###
+ ````swift
+ guard let myQueryDict: [String: String] = try? QueryEncoder().encode(query) else {
+     print("Failed to encode query to [String: String]")
+     return
+ }
+ ````
+ */
 public class QueryEncoder: Coder, Encoder {
 
+    /**
+     A `[String: String]` dictionary.
+     */
     private var dictionary: [String: String]
 
+    /**
+     The coding key path.
+     
+     ### Usage Example: ###
+     ````swift
+     let fieldName = Coder.getFieldName(from: codingPath)
+     ````
+     */
     public var codingPath: [CodingKey] = []
 
+    /**
+     The coding user info key.
+     */
     public var userInfo: [CodingUserInfoKey: Any] = [:]
 
+    /**
+     Initializer with a dictionary.
```

Comment:
It isn't obvious what this means as there isn't a dictionary passed in, is it initializing an empty dictionary? If so we should say what it's doing.